src/App.tsx

```diff
@@ -7,218 +7,91 @@ import { Filter } from "./panelSections/Filter";
+    [parameter.name]: finalValue
+  })
+}
+
+const getMergedParamStateForCallback = (state:ParamState, setParamState, monologueController:MonologueController) => (parameter: Parameter) => (finalValue: number) => {
+  monologueController.setParameter(parameter, finalValue);
+
+  setParamState({
+    ...state,
+    [parameter.name]: finalValue
+  })
+};
+
+const flushStateToMonologue = (state:ParamState, monologueController:MonologueController) => {
+  Object.keys(state).forEach(key => {
+    const { parameter, value } = state[key];
+    console.log(`Setting ${parameter.name} to ${value}`);
+    monologueController.setParameter(parameter, value);
+  })
+}
+
+const App = (props: AppProps) => {
+  const {
+    korgProgramDump,
+    monologueController,
+  } = props;
 
-const App = (props: KorgProgramDump) => {
   const selectPatch = (patch: KorgProgramDump) => {
-    // MASTER
     setPatchName(patch.patchName);
-    setVco1Octave(patch.oscilators[0].octave.value);
-    setDrive(patch.drive.value);
-    // VCO1
-    setVco1Octave(patch.oscilators[0].octave.value);
-    setVco1Wave(patch.oscilators[0].wave.value);
-    setVco1Shape(patch.oscilators[0].shape.value);
-    // VCO2
-    setVco2Octave(patch.oscilators[1].octave.value);
-    setVco2Wave(patch.oscilators[1].wave.value);
-    setVco2Pitch(patch.oscilators[1].pitch.value);
-    setVco2Duty(patch.oscilators[1].duty.value);
-    setVco2Shape(patch.oscilators[1].shape.value);
-    // MIX
-    setVco1Level(patch.oscilators[0].level.value);
-    setVco2Level(patch.oscilators[1].level.value);
-    // FILTER
-    setCutoff(patch.filter.cutoff.value);
-    setResonance(patch.filter.resonance.value);
-    // ENVELOPE
-    setEnvType(patch.envelope.type.value);
-    setEnvAttack(patch.envelope.attack.value);
-    setEnvDecay(patch.envelope.decay.value);
-    setEnvIntensity(patch.envelope.intensity.value);
-    setEnvTarget(patch.envelope.target.value);
-    // ENVELOPE
-    setLfoWave(patch.lfo.wave.value);
-    setLfoMode(patch.lfo.mode.value);
-    setLfoRate(patch.lfo.rate.value);
-    setLfoIntensity(patch.lfo.intensity.value);
-    setLfoTarget(patch.lfo.target.value);
-  };
 
-  const connectMidi = () => {
-    WebMidi.enable({ sysex: true })
-      .then(onEnabled)
-      .catch((err) => alert(err + "could not connect to midi"));
-  };
-  function onEnabled() {
-    setChannelIn(WebMidi.inputs[0].channels[1]);
-    setChannelOut(WebMidi.outputs[0].channels[1]);
-    WebMidi.inputs.forEach((input) =>
-      console.log(input.manufacturer, input.name)
-    );
-    WebMidi.outputs.forEach((output) =>
-      console.log(output.manufacturer, output.name)
-    );
-  }
+    const state = initialiseParamState(patch);
 
-  const listenSysex = () => {
-    channelIn.addListener("midi", (e) => console.log(e), {
-      duration: 4000,
-    });
+    setParamState(state);
+    flushStateToMonologue(state, monologueController);
   };
-  const listen = () => {
-    channelIn.addListener("midimessage", (e) => {
```

Comment:
This mapping code has been refactored out into two different places. The code that sets the values is in midi.ts, but the mapping that maps if a param is a linear pot or multi-pole is now a property of the Parameter itself.